### PR TITLE
Feat/recaps in identity

### PR DESCRIPTION
**Pull request by devceline**
- Export helpers from `did-jwt`
- Add registration via recaps

### CODE REVIEW

File: misc/identity-keys/src/identity-keys.ts [hunk 1]
```diff
@@ -52,6 +53,40 @@ export class IdentityKeys implements IIdentityKeys {
     return this.identityKeys.keys.includes(account);
   }
 
+  public async prepareRegistrationViaRecaps({
+    domain,
+    recapObject
+  }: {
+    domain: string
+    recapObject: {
+      att: Record<string,any>
```
Comment (chris13524) [inline misc/identity-keys/src/identity-keys.ts:62]:
```suggestion
      att: Record<string, Record<string, any[]>
```

File: misc/identity-keys/src/identity-keys.ts [hunk 2]
```diff
@@ -52,6 +53,40 @@ export class IdentityKeys implements IIdentityKeys {
...
+    }
+  }) {
+    const encodedRecap = objectToHex(recapObject);
```
Comment (chris13524) [inline misc/identity-keys/src/identity-keys.ts:65]: This should be unpadded base64url not hex. Maybe use `makeBase64UrlSafe()`?

File: misc/identity-keys/src/identity-keys.ts [hunk 3]
```diff
@@ -52,6 +53,40 @@ export class IdentityKeys implements IIdentityKeys {
...
+    const authRecap = `urn:recap:${encodedRecap}`
+
+    const { privKeyHex, pubKeyHex, privateKey } = await this.generateIdentityKey();
+    const didKey = encodeEd25519Key(privKeyHex);
+
+    const uri = `${domain}?walletconnect_identity_token=${didKey}`
```
Comment (chris13524) [inline misc/identity-keys/src/identity-keys.ts:71]: This needs to include the scheme. I think we should change `domain` to be `url: URL` and then parse the domain out of it and add the query param here with a URL builder.

Also change `walletconnect_identity_token` to `walletconnect_identity_key`

**Review by chris13524 — CHANGES_REQUESTED** [review]

File: misc/identity-keys/src/identity-keys.ts (hunk 2 above)
Comment (devceline) [reply misc/identity-keys/src/identity-keys.ts:65]: I know the naming is off but that's essentially what it is I made sure it produces the same string bartek's implementation did

File: misc/identity-keys/src/identity-keys.ts [hunk 4]
```diff
@@ -52,6 +53,40 @@ export class IdentityKeys implements IIdentityKeys {
     return this.identityKeys.keys.includes(account);
   }
 
+  public async prepareRegistrationViaRecaps({
```
Comment (chris13524) [inline misc/identity-keys/src/identity-keys.ts:56]: Less obscure word

```suggestion
  public async prepareRegistrationWithRecaps({
```

**Review by chris13524 — APPROVED** [review]
LGTM

**Pull request merged**